examples/esp_idf/stream/pytest/test_sample.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2024 Golioth, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import contextlib
+import logging
+import pytest
+import time
+import yaml
+import datetime
+import re
+
+LOGGER = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.anyio
+
+async def test_stream(board, device):
+    time.sleep(2)
+
+    # Set Golioth credential
+    golioth_cred = (await device.credentials.list())[0]
+    board.set_golioth_psk_credentials(golioth_cred.identity, golioth_cred.key)
```

Comment:
Not a blocker for this PR, but would be good to use the same method for `examples/zephyr/stream` as well (instead of manually calling `settings set golioth/...` commands, since we already have abstraction for that. Just to keep implementation differences as minimal as possible.

Reply:
Another question: what about settings WiFi credentials? Is ESP-IDF port different in this regard compared to Zephyr? (In Zephyr's pytest script there is an explicit configuration of WiFi credentials)

Reply:
> would be good to use the same method for examples/zephyr/stream as well

These functions are part of our `pytest-hil` pytest plugin, which isn't compatible with the Twister pytest plugins, so we can't use them for the Zephyr tests.

> what about settings WiFi credentials? Is ESP-IDF port different in this regard compared to Zephyr?

WiFi credentials are set by the `pytest-hil` plugin's `board` fixture. The `board` fixture doesn't set PSK credentials because those are generate at run-time by the test using the `python-golioth-tools` pytest plugin's `device` fixture.

Reply:
Thanks for explanation!